Wikipedia/Code/WMFArticleViewController.m

```diff
@@ -1762,8 +1762,9 @@ - (void)commitViewController:(UIViewController *)viewControllerToCommit {
                                      }
                                  }];
 
+    WMFArticle *wmfarticle = [self.previewStore itemForURL:self.articleURL];
```

Comment:
I'd suggest dataStore `fetchArticleForURL` (even though it's poorly named) since it caches the results.